lib/Doctrine/ODM/MongoDB/Aggregation/Stage/MatchStage.php

```diff
@@ -107,6 +107,15 @@ public function all(array $values): self
         return $this;
     }
 
+    /**
+     * Create a new Expr instance that can be used to build partial expressions
+     * for other operator methods.
+     */
+    public function createExpr(): Expr
```

Comment:
I'd suggest renaming this to `createQueryExpression` for better distinction from `createAggregationExpression`.

```suggestion
    public function createQueryExpression(): Expr
```